llvm/lib/Target/RISCV/RISCVInstrInfo.cpp

```diff
@@ -1918,6 +1999,78 @@ static void combineFPFusedMultiply(MachineInstr &Root, MachineInstr &Prev,
   DelInstrs.push_back(&Root);
 }
 
+// Combine (sh3add Z, (add X, (slli Y, 5))) to (sh3add (sh2add Y, Z), X).
```

Comment:
This comment doesn't match the implementation I think.

Reply:
What's wrong with it?

Reply:
`(sh3add Z, (add X, (slli Y, 5))) to (sh3add (sh2add Y, Z), X)` is too specific to the implementation below.